lib/src/widgets/common/common_progress_dialog.dart

```diff
@@ -8,13 +8,17 @@ Widget _defaultProgressWidget = Image.asset(Images.doubleRingLoader);
 
 class CommonProgressDialog {
   static ProgressDialog? _dialog;
+  static BuildContext? _currentContext;
 
   static Future<bool> show(
     BuildContext context, {
     String message = 'Please Wait !',
     bool isDismissible = false,
-  }) {
-    _dialog ??= ProgressDialog(context, isDismissible: isDismissible)
+  }) async {
+    await hide(); // Ensure any existing dialog is properly hidden
```

Comment:
Here i think, if existing dialog is shown, we must return directly.
But it would cause issue if isDismissible is true.

Need to discuss this on call once